packages/platform-browser/src/dom/dom_renderer.ts

```diff
@@ -100,8 +102,15 @@ export class DomRendererFactory2 implements RendererFactory2 {
         (<EmulatedEncapsulationDomRenderer2>renderer).applyToHost(element);
         return renderer;
       }
-      case ViewEncapsulation.Native:
+      case 1:
       case ViewEncapsulation.ShadowDom:
+        // TODO: remove the Native fallback logic in v12.
+        if (!hasLoggedNativeEncapsulationWarning && type.encapsulation === 1) {
+          hasLoggedNativeEncapsulationWarning = true;
+          console.warn(
+              'ViewEncapsulation.Native is no longer supported. Falling back to ShadowDom. The fallback will be removed after v12.');
```

Comment:
```suggestion
              'ViewEncapsulation.Native is no longer supported. Falling back to ViewEncapsulation.ShadowDom. The fallback will be removed in v12.');
```

May be we can also use `ngDevMode` here to tree-shake away this error message for prod bundles?

Reply:
I think CLI uses `ngDevMode` for both VE and Ivy (for AOT mode), we'd just need to make sure we are not accessing a global that was not defined (for JIT cases), i.e. `typeof ngDevMode !== 'undefined' && ...`.